src/app/components/book-list/book-list.component.ts

```diff
@@ -1,9 +1,96 @@
-import { Component, ChangeDetectionStrategy } from '@angular/core';
+import { Component, ChangeDetectionStrategy, OnInit } from '@angular/core';
+import { BooksService } from '@app/shared/services/books.service';
+import { ReadBook } from '@app/shared/models/read-book.model';
+import { map } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+import { MatDialog } from '@angular/material/dialog';
+import { EditDialogComponent } from '../book-dialog/edit/edit-dialog.component';
+import { ConfirmDialogComponent } from '../book-dialog/confirm/confirm-dialog.component';
+import { AddDialogComponent } from '../book-dialog/add/add-dialog.component';
+import { FormActions } from '@app/shared/enums';
 
 @Component({
   selector: 'app-book-list',
   templateUrl: './book-list.component.html',
   styleUrls: ['./book-list.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class BookListComponent {}
+export class BookListComponent implements OnInit {
+
+    books: Observable<ReadBook[]>;
+
+    constructor(private booksService:BooksService, public dialog: MatDialog){}
+
+    getBooks(): void {
```

Comment:
This method don't returns books, this method initializes books. Pls use more meaningful name.